include/cuco/detail/static_set/kernels.cuh

```diff
@@ -89,6 +90,257 @@ CUCO_KERNEL void find(InputIt first, cuco::detail::index_type n, OutputIt output
   }
 }
 
+template <typename CG,
+          typename Size,  // have to do this due to https://github.com/NVIDIA/cccl/issues/1523
+          typename ProbeKey,
+          typename Key,
+          typename AtomicT,
+          typename OutputIt1,
+          typename OutputIt2>
+__device__ void flush_buffer(CG const& tile,
+                             Size buffer_size,
+                             cuco::pair<ProbeKey, Key>* buffer,
+                             AtomicT* counter,
```

Comment:
```suggestion
                             cuda::atomic<Size, whateverthreadscope>& counter,
```

This way you can get rid of the `AtomicT` tparam and also make sure we pass a reference to a valid object instead of an arbitrary ptr.